lib/floki/raw_html.ex

```diff
@@ -133,7 +133,7 @@ defmodule Floki.RawHTML do
          self_closing_tags,
          line_ending
        ) do
-    encoder =
+    curr_encoder =
```

Comment:
There's another bug here, the identity encoder is also being applied to the script/style/title attributes.
This encoder change should be done inside of `case children` to avoid issues, and will also fix the bug you found.

Reply:
Ah you're right! I've updated the test and moved that check